packages/main/src/MenuItem.ts

```diff
@@ -280,10 +341,68 @@ class MenuItem extends ListItem implements IMenuItem {
 		return this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
 	}
 
+	get _markSelected() {
+		return this.isSelected && this._itemSelectionMode !== ItemSelectionMode.None;
+	}
+
+	/** Returns menu item groups */
+	get _menuItemGroups() {
+		return this.items.filter((item) : item is MenuItemGroup => item.isGroup);
+	}
+
+	/** Returns menu items */
 	get _menuItems() {
 		return this.items.filter((item): item is MenuItem => !item.isSeparator);
 	}
 
+	/** Returns all menu items (including those in groups */
+	get _allMenuItems() {
+		const items: MenuItem[] = [];
+
+		this.items.forEach(item => {
+			if (item.isGroup) {
```

Comment:
`item.isGroup` is not a proper typescript check, therefore you need `as MenuItemGroup` on the next line. Same for other parts of the code. The correct way to do it is for example as done in Side Navigation:

```
	get isSideNavigationItemBase() {
		return true;
	}

...............

const isInstanceOfSideNavigationItemBase = (object: any): object is SideNavigationItemBase => {
	return "isSideNavigationItemBase" in object;
};
```

You must make a similar instanceof function for MenuItem/Group etc... and then when you do the same check as on line 363, Typescript will know that if the check passed, the type of your item is whatever you put in `is`, as shown above.